src/main/scala/info/bethard/anafora/Anafora.scala

```diff
@@ -34,18 +36,37 @@ abstract class Annotation(val xml: Elem) {
 object Entity {
   def apply(xml: Elem) = new Entity(xml)
 }
-
 class Entity(xml: Elem) extends Annotation(xml) {
   lazy val spans: IndexedSeq[(Int, Int)] =
     (xml \ "span" \ ElemText).flatMap(_.split(";")).map(_.split(",").map(_.toInt) match {
       case Array(start, end) => (start, end)
     }).sorted
   lazy val fullSpan: (Int, Int) = (spans.map(_._1).min, spans.map(_._2).max)
-
   def text(implicit data: Data): Option[String] = data.text.map(text => spans.map {
     case (start, end) => text.substring(start, end)
   }.mkString("..."))
-
+  private def recursiveSpan(entity: Entity, start: Int, end: Int)(implicit data:Data): (Int, Int) = {
+    var newStart: Int = start
+    var newEnd: Int = end
+    for (result <- entity.xml  \ "properties" \ * \ ElemText map (p => data.entities filter (e => e.id == p))) {
+      if (result.length > 0) for (childEntity <- result) {
+        newStart = List(newStart, childEntity.fullSpan._1).min
+        newEnd = List(newEnd, childEntity.fullSpan._2).max
+        recursiveSpan(childEntity, newStart, newEnd) match {
+          case (x, y) => newStart = x; newEnd = y
+          case _ =>
+        }
+      }
+    }
+    (newStart, newEnd)
+  }
+  def expandedSpan(implicit data: Data): (Int, Int) = recursiveSpan(this, fullSpan._1, fullSpan._2)
```

Comment:
I think this can be rewritten as something like:
```
  def expandedSpan(implicit data: Data): (Int, Int) = {
    val allSpans = entityDescendants.flatMap(_.spans)
    (allSpans.map(_._1).min, allSpans.map(_._2).max)
  }
```
Right?